spec/shoryuken/polling/weighted_round_robin_spec.rb

```diff
@@ -106,12 +106,30 @@
   end
 
   describe '#message_processed' do
-    it 'removes paused queue, adds to active queues' do
-      strategy = Shoryuken::Polling::WeightedRoundRobin.new([queue1, queue2])
-      strategy.send(:pause, queue1)
-      expect(strategy.active_queues).to eq([[queue2, 1]])
-      strategy.message_processed(queue1)
-      expect(strategy.active_queues).to eq([[queue2, 1], [queue1, 1]])
+    it 'removes delay from paused queue' do
+      queues << queue1
+      queues << queue2
+      subject.send(:pause, queue1)
+
+      expect(subject.active_queues).to eq([[queue2, 1]])
+      subject.message_processed(queue1)
+      expect(subject.active_queues).to eq([[queue2, 1]])
```

Comment:
I believe to keep the intent on #644 the test should stay the same here:

```
expect(strategy.active_queues).to eq([[queue2, 1], [queue1, 1]])
```

Reply:
So the behavior of `message_processed` for `WeightedRoundRobin` changed slightly that it doesn't actually unpause the queue immediately, it now just changes the "paused until" timestamp such that when the next queue is requested, `unpause_queues` will actually do the unpausing. I think the intend of #644 is still preserved, it's just that the internal data structures are modified at slightly different times.

Let me know if I'm misunderstanding it.

I think the new proposed change to `WeightedRoundRobin#message_processed` simplifies things a little since now only `unpause_queues` is responsible for manipulating the right data structures for unpausing.

Reply:
Got it. I think this is a good implementation. It does simplify the responsibility for unpausing queues 💯